opensearch-operator/api/v1/opensearch_types.go

```diff
@@ -126,6 +126,9 @@ type NodePool struct {
 	PriorityClassName         string                            `json:"priorityClassName,omitempty"`
 	Pdb                       *PdbConfig                        `json:"pdb,omitempty"`
 	Probes                    *ProbesConfig                     `json:"probes,omitempty"`
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +kubebuilder:validation:Type=object
```

Comment:
Even after adding these two markers, the full CRD is generated instead of short one like this
`sidecarContainers:
  type: array
  x-kubernetes-preserve-unknown-fields: true
  items:
    type: object
    x-kubernetes-preserve-unknown-fields: true`
    
    Here is the error that i am seeing while running the test
    `          msg: "unable to install CRDs onto control plane: unable to create CRD instances: unable to create CRD \"opensearchclusters.opensearch.opster.io\": CustomResourceDefinition.apiextensions.k8s.io \"opensearchclusters.opensearch.opster.io\" is invalid: [spec.validation.openAPIV3Schema.properties[spec].properties[nodePools].items.properties[sidecarContainers].items.properties[ports].items.properties[protocol].allOf[0].default: Forbidden: must be undefined to be structural, spec.validation.openAPIV3Schema.properties[spec].properties[nodePools].items.properties[sidecarContainers].items.properties[ports].items.properties[protocol].allOf[1].default: Forbidden: must be undefined to be structural, spec.validation.openAPIV3Schema.properties[spec].properties[nodePools].items.properties[sidecarContainers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property]",
`

Reply:
When recursive validation is not possible, need to use schemaless
```suggestion
	// +kubebuilder:pruning:PreserveUnknownFields
	// +kubebuilder:validation:Schemaless
```

Reply:
thanks, it worked!!